### PR TITLE
Add variable search-engine-config-list in order to support customization

### LINKED ISSUE #7999: Can't add new search engines
#### Description :octocat:
Can't add new search engines using the instructions in the search-engine layer.
The "added" engine shows up in the list but when you select it the following error appears:

`Wrong type argument: commandp, engine/search-custom1`

assuming i added an engine named `custom1`

#### Reproduction guide :beetle:
- Start Emacs
- install search-engine layer
- add the following to the dotspacemacs/user-config

  ```elsip
  (push '(custom1
          :name "Custom Search Engine 1"
          :url "http://www.domain.com/s/stuff_sutff_remember_to_replace_search_candidate_with_%s")
        search-engine-alist)
  ```
- restart emacs
- `SPC a /` (observe `Custom Search Engine 1` shows up in the list)
- select the custom1 option (observe the `Wrong type argument: commandp, engine/search-custom1` message)

*Observed behaviour:* :eyes: :broken_heart:
`Wrong type argument: commandp, engine/search-custom1` when selecting custom added search engine

*Expected behaviour:* :heart: :smile:
can use the custom added engine just like the ones there by default

(%LAST_KEYS%)
%SYSTEM_INFO%

#### Backtrace :paw_prints:
```
Wrong type argument: commandp, engine/search-custom1
```

### PR DESCRIPTION
**Pull request by haishengwu-okta**
My attempt to fix #7999

### CODE REVIEW

**Comment by agzam** [comment]
Does this let you use other keys in `defengine` like `:browser` and `:term-transformation-hook`?

**Comment by haishengwu-okta** [comment]
@agzam no I didn't take in account that in this PR.

File: layers/+web-services/search-engine/packages.el [hunk 1]
```diff
@@ -73,6 +73,7 @@
               (wolfram-alpha
                :name "Wolfram Alpha"
                :url "https://www.wolframalpha.com/input/?i=%s")))
+      (setq search-engine-alist (append search-engine-alist search-engine-config-list))
```
Comment (smile13241324) [inline layers/+web-services/search-engine/packages.el]: Nice fix, however you should not invoke `setq` a second time. Rather use the backtick notation and add the content of `search-engine-config-list` directly at creation time to `search-engine-alist`

Here is a minimal example how to do this:
```elisp
  (setq mmm '(1 2 3))
  (setq myList
        `(Bla ;;; Note the backtick ` this is no quote '
          bla
          ,@mmm)) ;;; values starting with , will be evaluated. @ means the content of the list shall be inserted here.
```

Comment (duianto) [reply layers/+web-services/search-engine/packages.el]: If this is a new concept, then the Emacs manuals explanation might also be helpful:
https://www.gnu.org/software/emacs/manual/html_node/elisp/Backquote.html

**Comment by haishengwu-okta** [comment]
@smile13241324 address your comments. thank you for reviewing!
(the force-push is because I did rebase `develop` branch with latest changes.)

**Comment by smile13241324** [comment]
Thank your for contributing to spacemacs :purple_heart:. I have cherry-picked your commits into develop. To do so I have squashed them. You can safely remove your local branch now.

**Pull request closed without merging**